geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/distributed/WaitUntilFlushedFunction.java

```diff
@@ -85,8 +72,10 @@ public void execute(FunctionContext context) {
       }
 
     } else {
-      throw new IllegalArgumentException(
+      IllegalStateException illegalStateException = new IllegalStateException(
           "The AEQ does not exist for the index " + indexName + " region " + region.getFullPath());
+      logger.error(illegalStateException.getMessage());
```

Comment:
@jhuynh1 
If we throw exception from WaitUntilFlushedFunction then executing side will not get a true or false.
Unless we send the result explicitly using resultSender.lastResult(result);
or other alternative is to send this exception using sendException() Api and executing side needs to handle this exception to return a boolean result for waitUntilFlushed.

@upthewaterspout @dschneider-pivotal @bschuchardt - please correct me if I am wrong.